From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Robin=20Lind=C3=A9n?= <dev@robinlinden.eu>
Date: Sat, 21 Sep 2019 17:38:41 +0200
Subject: [PATCH 1/2] Get libsodium with FetchContent

---
 CMakeLists.txt           | 2 +-
 cmake/Dependencies.cmake | 9 ++++++++-
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c93f17e6..d7a824a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -156,7 +156,7 @@ set(toxcore_SOURCES ${toxcore_SOURCES}
 include(CheckFunctionExists)
 check_function_exists(explicit_bzero HAVE_EXPLICIT_BZERO)
 check_function_exists(memset_s HAVE_MEMSET_S)
-set(toxcore_LINK_MODULES ${toxcore_LINK_MODULES} ${LIBSODIUM_LIBRARIES})
+set(toxcore_LINK_MODULES ${toxcore_LINK_MODULES} sodium)
 set(toxcore_PKGCONFIG_REQUIRES ${toxcore_PKGCONFIG_REQUIRES} libsodium)
 
 # LAYER 2: Basic networking
diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 5970fde7..15fe550f 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -4,6 +4,7 @@
 #
 ###############################################################################
 
+include(FetchContent)
 include(ModulePackage)
 
 find_package(Threads REQUIRED)
@@ -13,7 +14,13 @@ find_library(RT_LIBRARIES           rt           )
 find_library(SOCKET_LIBRARIES       socket       )
 
 # For toxcore.
-pkg_use_module(LIBSODIUM            libsodium    )
+FetchContent_Declare(Sodium
+    GIT_REPOSITORY https://github.com/robinlinden/libsodium-cmake.git
+    GIT_TAG 90d87cdb228ee3ea5d50449d738b4ba264c27f2b # 1.0.17
+)
+set(SODIUM_DISABLE_TESTS ON CACHE INTERNAL "")
+set(SODIUM_PRETEND_TO_BE_CONFIGURED ON CACHE INTERNAL "")
+FetchContent_MakeAvailable(Sodium)
 
 # For toxav.
 pkg_use_module(OPUS                 opus         )
-- 

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Robin=20Lind=C3=A9n?= <dev@robinlinden.eu>
Date: Sat, 21 Sep 2019 20:02:13 +0200
Subject: [PATCH 2/2] Export tox/tox.h as part of the toxcore target

---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d7a824a2..e3f6288f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -336,6 +336,14 @@ make_version_script(toxcore ${toxcore_API_HEADERS})
 # "${CMAKE_INSTALL_INCLUDEDIR}/tox".
 install_module(toxcore DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tox)
 
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/tox)
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/toxcore/tox.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/include/tox/tox.h
+  COPYONLY
+)
+target_include_directories(toxcore_static PUBLIC include)
+
 ################################################################################
 #
 # :: Unit tests: no networking, just pure function calls.
-- 

